ops/log.py

```diff
@@ -29,15 +29,16 @@ def emit(self, record):
 def setup_root_logging(model_backend, debug=False):
     """Setup python logging to forward messages to juju-log.
 
+    By default, logging is set to DEBUG level, and messages will be filtered by Juju.
+    Charmers can also set their own default log level with::
+
+      logging.getLogger().setLevel(logging.INFO)
+
     model_backend -- a ModelBackend to use for juju-log
-    debug -- if True, enable DEBUG level logging, and write logs to stderr as well as to juju-log.
+    debug -- if True, write logs to stderr as well as to juju-log.
```

Comment:
This sounds reasonable in general. Two questions:

1) Should we check the juju version and not spam people unless we are in a version that has fixed logs not to send debugging to people, so that we're not holding ourselves back from writing down debug statements in the code to prevent showing them to the general user? 

2) Is there a context in which we don't want to log debug when running outside juju?  I guess in tests it makes sense as well. Does it make sense when debugging in the shell? I guess it does as well. Any others?

Reply:
1. I don't think so. Juju has always had the ability to filter late with "juju debug-log --level=INFO", so if things are spamming, there are ways to not see it. Reactive itself is very chatty at DEBUG level (see [bug #1874000](https://bugs.launchpad.net/juju/+bug/1874000) for examples). So people will already be used to filtering.

Juju has also always supported people applying record-time filtering with 'juju model-config logging-config unit=INFO'. It is just that the default log level for unit messages was DEBUG.

So I think anyone bothered by verbosity already has knobs to handle it, and it is simpler and more obvious if we leave it to the existing tools. Trying to second guess them means when they really do want DEBUG, they'll find it hard to enable.

2. We have [bug #1861987](https://bugs.launchpad.net/juju/+bug/1861987) tracking the desire to have the juju model config for a unit exposed to the charm, so that they can chose to update the default logging. (So we don't have to exec a command if it is going to ignore the result.) In the short term, it isn't a correctness thing, just an efficiency one.

I would be happy for us to have something like an env var (juju uses JUJU_LOGGING_CONFIG) or a standard commandline parameter for main.main(). But both of those feel like a stretch in search of a use case right now, rather than a concrete need.

Reply:
:+1: